docs/snapshot-testing.md

```diff
@@ -0,0 +1,113 @@
+# Snapshot Testing
+
+## What is Snapshot Testing?
+
+Snapshot testing is an extension of usual expected-vs-actual assertion where the following conditions apply:
+
+1. The expected object ("snapshot" afterwards) is too large to be contained in a few lines of code (e.g. a web page HTML content)
+1. The snapshot should be tolerant to some degree of data variation (e.g. date and time of test execution)
+1. (Optional) There is an automated way to update the snapshot based on the state of the actual object at any point of time
+
+In the case of TEAMMATES, snapshot testing is useful for the following comparisons:
+
+1. Actual rendered web page vs expected web page content/DOM structure
+1. Actual email generated by the system vs expected email content
+1. Actual CSV generated by the system vs expected CSV content
+
+## How does Snapshot Testing work?
+
+When testing (i.e. doing the assertions), snapshot testing is the same as normal assertion.
```

Comment:
Maybe change this to:-
"Snapshots tests can run in two modes, namely, verification and auto-update. 

The verification mode is essentially the same as a normal assertion, where the rendered web page is compared against the expected web page content"

so that there is no confusion that there are two modes and they have very different features.